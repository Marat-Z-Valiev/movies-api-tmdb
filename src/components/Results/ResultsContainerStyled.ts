import styled from "styled-components";
import fadeInAnimation from "../../animation/fadeInAnimation";

const ResultsContainerStyled = styled.div`
	display: grid;
	justify-content: center;
	grid-template-columns: repeat(auto-fill, minmax(333px, 1fr));
	grid-gap: 40px 20px;
	padding: 30px;
	grid-template-rows: 1fr min-content;
	${fadeInAnimation};

	@media (min-width: 320px) and (max-width: 425px) {
		display: block;
	}
`;

export default ResultsContainerStyled;
