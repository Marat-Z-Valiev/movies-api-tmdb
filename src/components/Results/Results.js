import React from "react";
import ResultItem from "../ResultItem/ResultItem";
import ResultsContainerStyled from "./ResultsContainerStyled";
import PropTypes from "prop-types";

const Results = ({totalResults, results}) => {
	const sortedArray = results
		.slice(0, results.length)
		.sort((a, b) =>
			a.release_date && b.release_date
				? b.release_date.split("-")[0] - a.release_date.split("-")[0]
				: ""
		);

	return (
		<>
			{totalResults ? (
				<p style={{textAlign: "center", color: "#ffffff", fontSize: "1.5rem"}}>
					Total results is {totalResults}
				</p>
			) : (
				""
			)}
			<ResultsContainerStyled>
				{Object.keys(sortedArray).map((key) => (
					<ResultItem className="result-tile" key={key} result={results[key]} />
				))}
			</ResultsContainerStyled>
		</>
	);
};

Results.propTypes = {
	totalResults: PropTypes.number,
	results: PropTypes.array.isRequired,
};

export default Results;
