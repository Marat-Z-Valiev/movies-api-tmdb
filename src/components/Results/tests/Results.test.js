import React from "react";
import {shallow, configure, mount} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Results from "../Results";

configure({adapter: new Adapter()});

describe("Results", () => {
	const props = {
		totalResults: 100,
		results: [],
	};

	it("should render correctly", () => {
		const wrapper = mount(<Results {...props} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
