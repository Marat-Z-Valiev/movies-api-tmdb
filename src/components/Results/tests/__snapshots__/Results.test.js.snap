// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render correctly 1`] = `
<Results
  results={Array []}
  totalResults={100}
>
  <p
    style={
      Object {
        "color": "#ffffff",
        "fontSize": "1.5rem",
        "textAlign": "center",
      }
    }
  >
    Total results is 
    100
  </p>
  <ResultsContainerStyled>
    <div
      className="ResultsContainerStyled-sc-1p24i8x-0 kFiqOW"
    />
  </ResultsContainerStyled>
</Results>
`;
